Replace black, isort and flake8 with ruff

thekaranacharya opened the pull request:
- Gets rid of black, isort and flake8, and replaces them with ruff
- Updates the pyproject.toml with ruff settings
- Updates the makefile: `make lint` and `make autoformat` with the correct commands
- Remove flake8 config file
- Using the default config from [here](https://docs.astral.sh/ruff/configuration/#__tabbed_1_1) + existing ignores from flake8 and isort
- Add ruff linting + formatting as a pre-commit hook
- Other files in this PR that changed are the transformed files after linting + formatting

thekaranacharya committed:
Remove flake8 file

thekaranacharya committed:
Update makefile

thekaranacharya committed:
Update pyproject and poetry lock

thekaranacharya committed:
Update job name

thekaranacharya committed:
Replace flake8 with ruff

thekaranacharya committed:
New lint and format fixes

thekaranacharya committed:
Restore removed imports, and add them to all

thekaranacharya committed:
Remove from all, just add ignore

thekaranacharya committed:
Fix autolint error

thekaranacharya committed:
Update lock file

thekaranacharya committed:
Re-place type ignore

thekaranacharya committed:
Add pre-commit config yaml and installed hook

thekaranacharya commented:
@CalebCourier @zsimjee I added the pre commit config yaml and installed the hook with `pre-commit install` - that installed the hook in my local guardrails/.git, which is not synced up. What's the best way to have that pre-commit hook here? 

- Should we add a command inside our `make precommit` [here](https://github.com/guardrails-ai/guardrails/blob/6de5641b8f269164cd57cd95f32dacb9e7d83537/Makefile#L83) to run `pre-commit install`?

zsimjee commented on .flake8:
```diff
@@ -1,6 +0,0 @@
-[flake8]
-count = True
-statistics = True
-max-line-length = 88
```
let's make sure max line length is still respected in the new ruff config

thekaranacharya replied:
Check the main comment (https://docs.astral.sh/ruff/configuration/#__tabbed_1_1): if not specified ruff uses 88 by default

zsimjee commented on .flake8:
```diff
@@ -1,6 +0,0 @@
-[flake8]
-count = True
-statistics = True
```
what's the new way to get count + stats?

thekaranacharya replied:
Count according to flake8 [here](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-count) refers to printing the number of errors. Ruff, by default prints that

thekaranacharya replied:
For statistics, seems like the same thing. This is how a ruff sample output looks like:
```
poetry run ruff check guardrails/ tests/
guardrails/demo.py:1:8: F401 [*] `json` imported but unused
guardrails/demo.py:3:8: F401 [*] `openai` imported but unused
guardrails/demo.py:10:5: F841 Local variable `a` is assigned to but never used
guardrails/demo.py:11:5: F841 Local variable `b` is assigned to but never used
Found 4 errors.
[*] 2 fixable with the `--fix` option (2 hidden fixes can be enabled with the `--unsafe-fixes` option).
make: *** [lint] Error 1
```

Has both the total count and description for each error is shown

zsimjee commented:
> @CalebCourier @zsimjee I added the pre commit config yaml and installed the hook with `pre-commit install` - that installed the hook in my local guardrails/.git, which is not synced up. What's the best way to have that pre-commit hook here?
> 
> * Should we add a command inside our `make precommit` [here](https://github.com/guardrails-ai/guardrails/blob/6de5641b8f269164cd57cd95f32dacb9e7d83537/Makefile#L83) to run `pre-commit install`?

installing precommit should be included in make dev/full/all.

CalebCourier reviewed (dismissed):
LGTM!

thekaranacharya committed:
Add pre-commit install to `make dev`

thekaranacharya commented:
> > @CalebCourier @zsimjee I added the pre commit config yaml and installed the hook with `pre-commit install` - that installed the hook in my local guardrails/.git, which is not synced up. What's the best way to have that pre-commit hook here?
> > 
> > * Should we add a command inside our `make precommit` [here](https://github.com/guardrails-ai/guardrails/blob/6de5641b8f269164cd57cd95f32dacb9e7d83537/Makefile#L83) to run `pre-commit install`?
> 
> installing precommit should be included in make dev/full/all.

Added it to `make dev` - as makes sense to have them when we're adding new code

zsimjee committed:
pre-commit hook failure

zsimjee commented:
I pulled down the PR and tested the precommit hook, it doens't seem to work. I added a line with really long line, and it passed the hook

zsimjee commented:
but it worked on an unused variable!

thekaranacharya committed:
Add E501 check

thekaranacharya committed:
Update makefile

thekaranacharya committed:
Autoformat

thekaranacharya commented:
> I pulled down the PR and tested the precommit hook, it doens't seem to work. I added a line with really long line, and it passed the hook

@zsimjee Added the E501 check to test for line too long. Now tested in a fresh environment, added a new long line example in another file, the pre commit hook runs and fails - thereby stopping the commit!
